glidertools/plot.py

```diff
@@ -350,16 +353,30 @@ def bin_size(depth, bins=None, ax=None, add_colorbar=True, **hist_kwargs):
         if bins is None:
             binning_freq = 50
             ybins = get_optimal_bins(depth, binning_freq)[0]
+            xbins = r_[nan, diff(ybins)]
+
         else:
-            ybins = bins
-        xbins = r_[nan, diff(ybins)]
+            ybins = bins[1]
+            xbins = []
+            for k in range(len(ybins) - 1):
+                d0 = ybins[k]
+                d1 = ybins[k + 1]
+
+                i = (y > d0) & (y < d1)
+                xbins += (nanmedian(x[i]),)
+            xbins = r_[nan, xbins]
 
         if ax is None:
             fig, ax = subplots(1, 1, figsize=[4, 6])
+        if bins is None:
+            im = ax.hist2d(
+                x, y, bins=50, norm=LogNorm(), rasterized=True, **hist_kwargs
```

Comment:
if `bins` is defined in the previous if-else statement, then this second catch won't have to happen. Make the code a bit cleaner